### ISSUE TITLE
Warn if multiple instances of context provider/consumer are found

### ISSUE THREAD

**Issue by rtsao** [styletron/styletron#267]
See: https://github.com/facebook/react/issues/13346

In development, it would be nice to issue warnings at runtime if multiple instances can be detected (where communication won't work as expected).

**Comment by tajo** [comment]
We should also remove `create-react-context`, drop React <16.3 support and add it to the v5 release.

**Comment by tajo** [comment]
You should already see a warning describing this when Styletron context is missing due to multiple instances of React. Is there something else we could do?

**Comment by rtsao** [comment]
Yeah, for example, there could be multiple instances of styletron-react, but only a single version of react. This might happen with certain usages of `npm link` and equivalents.

In development, we could detect if there are multiple instances via some singleton/global variable.

**Comment by asherccohen** [comment]
Hi, Any news on this? 

I'm building a UI library with different themes and ONE generic storybook to handle them all, I see that when I make a change to one theme some settings "pollute" the other library.
I tried both having a shared context and by separating them.

Is there a way around this?
